src/hotspot/share/classfile/loaderConstraints.cpp

```diff
@@ -447,6 +443,18 @@ InstanceKlass* LoaderConstraintTable::find_constrained_klass(Symbol* name,
   return nullptr;
 }
 
+// Removes a class that was added to the table then class loading subsequently failed for this class.
+void LoaderConstraintTable::remove_failed_loaded_klass(InstanceKlass* klass,
```

Comment:
This only seems to null out the klass ptr not remove the constraint. ??

Reply:
I don't understand what nulling out the klass actually achieves. ?? It seems to me to be allowing for a future attempt to re-load the same named class in the same loader, but IIRC that should not be allowed - at least in terms of resolution errors.